lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -46,8 +48,25 @@ def fm_pkg_and_cmd_name
         %w[satellite-maintain satellite-maintain]
       end
 
+      def satellite_maintain_target_version
+        satellite_maintain_config['current_satellite_version']
+      end
+
+      def satellite_upgrade_allowed?
+        current_minor_version == satellite_maintain_config['previous_satellite_version'] ||
+          ForemanMaintain.upgrade_in_progress == satellite_maintain_target_version
+      end
+
       private
 
+      def satellite_maintain_config
+        if File.exist?(SATELLITE_MAINTAIN_CONFIG)
+          YAML.load_file(SATELLITE_MAINTAIN_CONFIG)
+        else
+          raise "Couldn't load satellite-maintain configuration file #{SATELLITE_MAINTAIN_CONFIG}."
```

Comment:
```suggestion
          raise "Could not load satellite-maintain configuration file #{SATELLITE_MAINTAIN_CONFIG}."
```

Contractions get messy